sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -730,14 +754,16 @@ def readinto(self, stream: IO[T]) -> int:
         # Write the content to the user stream if there is data left
         if self._offset < len(self._current_content):
             content = self._current_content[self._offset:]
-            stream.write(content)
+            stream.write(content)  # type: ignore [call-overload]
```

Comment:
Alright so in main stream is type-hinted as `IO[T]`. But in the instance I've tested with type `str` and `bytes`, we always load into a `BytesIO`.

Here is the MyPy complaint for when it is typehinted as `IO[T]`:
```
azure\storage\blob\_download.py:758: error: No overload variant of "write" of "IO" matches argument type "bytes"  [call-overload]
azure\storage\blob\_download.py:758: note: Possible overload variants:
azure\storage\blob\_download.py:758: note:     def write(self, str, /) -> int
```

So as an alternative, I tried setting it to `Union[BytesIO, StringIO]` since that sort of seems analogous to what we are looking for? And I am met with this:
`azure\storage\blob\_download.py:757: error: Argument 1 to "write" of "TextIOBase" has incompatible type "bytes"; expected "str"  [arg-type]`
![image](https://github.com/Azure/azure-sdk-for-python/assets/101599632/8e6fee6d-cab1-49b6-b337-0e7d1a6047e8)

In general, the confusion here is what these should be properly typed as, and Python doesn't really seem to have a great standard either.

IO[T] when T is defined vs. StringIO vs BytesIO vs IO[str] vs IO[bytes] ? ? ?

Reply:
I think this is a combination of the fact that the `stream` type hint is actually wrong (it's really `IO[bytes]` and we need to fix it but later) and the fact that MyPy can't deal with `T`. So ignore for now is fine.